### DISCUSSION TITLE
RAM usage increase by the time

### DISCUSSION THREAD

**Discussion by rajeshisnepali** [osm-search/Nominatim#2801] [Q&A]
**Describe the bug**
For instance, current (Sun Aug 14, 2:00 UTC) RAM usage is 3GB, later (Sun Aug14, 8:00 UTC) RAM usage is 3.1GB or can be more.  

**To Reproduce**
- Install nominatim using [latest docs](https://nominatim.org/release-docs/latest/appendix/Install-on-Ubuntu-20/) in a high configuration server
- Import North America map data [https://download.geofabrik.de/north-america-latest.osm.pbf](https://download.geofabrik.de/north-america-latest.osm.pbf)
- Backup database & migrate to a lower-end server using [https://nominatim.org/release-docs/latest/admin/Advanced-Installations/#moving-the-database-to-another-machine](https://nominatim.org/release-docs/latest/admin/Advanced-Installations/#moving-the-database-to-another-machine)
-  run nominatim in another (lower-end) server
- wait for few hours or days to see the increase in RAM.

**Software Environment (please complete the following information):**
- Nominatim version: Nominatim-4.0.1
- Postgresql version: PostgreSQL 12.11 
- Postgis version: I didn't find `HINT:  No function matches the given name and argument types. You might need to add explicit type casts.`
- OS: ubuntu 20.04

**Hardware Configuration (please complete the following information):**
- RAM: 8GB
- number of CPUs: 2
- type and size of disks: gp2 storage & 400GB
- bare metal/AWS/other cloud service: 

**Postgresql Configuration:**

 ```
    shared_buffers = 2GB
    maintenance_work_mem = 1GB
    autovacuum_work_mem = 2GB
    work_mem = 6MB
    effective_cache_size = 3GB
    synchronous_commit = off
    max_wal_size = 1GB
    checkpoint_timeout = 10min
    checkpoint_completion_target = 0.9
    autovacuum_max_workers = 1
```

**Additional context**

I recently noticed that issue after my app was down. And then monitored my RAM usage. 

![image](https://user-images.githubusercontent.com/23610013/184546958-61893faa-3519-409e-8c40-7b5e28f772f5.png)

**Comment by rajeshisnepali** [comment]
**Update**
![image](https://user-images.githubusercontent.com/23610013/184581538-1e17e242-05fa-41d3-83a7-57bdc08bca36.png)

**Reply by lonvia** [reply]
There is nothing unusual going on here. Postgresql loads more and more data with time which ends up in the buffer cache.

**Reply by rajeshisnepali** [reply]
So that means I've to manually restart my server periodically to clear the buffer cache?

**Reply by mtmail** [reply]
You can limit the size of the buffer https://www.postgresql.org/docs/12/runtime-config-resource.html

**Reply by rajeshisnepali** [reply]
![image](https://user-images.githubusercontent.com/23610013/185032452-112115b1-36d3-4466-80f9-3f86fb7f8d7f.png)
I don't think it is working.
